### PR TITLE
Milestone 2: User interface 

### PR DESCRIPTION
**Pull request by tanzila-abedin**
- Added display content 
- players input
- game flow 

ps. It is a non-functioning game at this point.

### CODE REVIEW

**Review by george-shammar — CHANGES_REQUESTED** [review]
**### STATUS: CHANGES REQUIRED :recycle: :recycle:** 

### **You have done a great job here. :muscle: 

**### It's impressive that you have included a README in your project though it is not required at this stage.** :clap: :clap: :clap:**

### **Below are the improvements required for this milestone:**

- [ ] **Remember to ask the players to select from available moves (Note: The board needs not to be updated at this stage.)**
- [ ] **Your game should inform a player if the selected move is invalid (without computing what an invalid move is). Please refer to this project line for clarity:** (https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)
- [ ] **After displaying a win (or tie), your game should loop for a new game except players end it. Game repeats all actions for the next player's move.**
- [ ] **There are 9 possible moves but yours currently ends at 8.** 

**Hint: Check your while loop. This might complicate your next milestone- game logic. You can use something like this:**

`game_on = true` to start the game.
**......while loop......**
`game_on = false` to end the game.

**Your game does not need to be fully functional for this milestone but the game flow should be fully thought through.**

**Request for another code review once you're ready.

Happy coding.

Reach me on slack on @george gbenle if you require further clarification.**

File: bin/main [hunk 1]
```diff
@@ -1,3 +1,96 @@
...
+      @arr << move
+      true
+
+    end
+  end
+
+  def players
+    turn = 0
+    while turn < 8
+      if turn.even?
```
Comment (Dipeshtwis) [inline bin/main:33]: In line 32, It should be `9` or `<=`

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @tanzila-abedin and @Stephanie041996 , I'm Dipesh your reviewer for this PR. 

It's good to see that you have implemented the loop and fixed some of the changes, but in order to approve, we have to work a little bit more. I have also left some comments.

## Always re-check if all the suggested changes applied or not.

## For Approval

- [x] There are 9 possible moves but yours currently ends at 8. I have suggested how to fix in the comment.
- [x] The game allows me to keep playing with an empty input. Please fix that. Write a validation so the game doesn't allow to continue playing if the input name is empty.

Screenshot (1286)_LI

## Let's fix the required change, and we will be done.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional question).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Github](https://github.com/Dipeshtwis)

File: bin/main [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+    else
+      @arr << move
+      true
+
+    end
+  end
+
+  def players
+    turn = 0
+    while turn <= 9
```
Comment (GzimAsani) [inline bin/main]: This in line 32, it should be either `while turn < 9` or `while turn <= 8`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: :red_circle: 

### Hello **team**, great job implementing all the required changes asked from a previous TSE, however we have to make one more change to fully complete this millstone:
You have failed to implement the following change asked from the previous reviewer
- [ ] _There are 9 possible moves but yours currently ends at 8. I have suggested how to fix in the comment_. With the new change your game has more than 9 rounds and I think you have confused the suggestion made by the previous reviewer. Check the comment to fully understand the requirement

_Please implement all the above mentioned requirements and ask for another code review whenver you are ready_

### **If you have any question related to the requirements please add a comment here and ping me on slack @GzimAsani**

HAPPY CODING :computer: :desktop_computer:

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @tanzila-abedin & @Stephanie041996, I'm Dipesh your reviewer for this PR. 

Good job on implementing the suggested changes👏👏

**You may close this Pull Request and merge the changes**
  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏

Code review by **Dipesh Kumar**

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Twitter](https://twitter.com/97deepeshkumar)

**Pull request merged**